src/WooFeed.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Netzstrategen\WoocommerceDynamicShippingInfo;
+
+/**
+ * Integration with Webappick Product Feed / CTX Feed plugin.
+ *
+ * Test feed generation on command line:
+ * ```
+ * wp eval '$name = "wf_config" . "google_de"; $config = get_option($name); woo_feed_generate_feed($config, $name);'
+ * ```
+ */
+class WooFeed {
+
+  /**
+   * Adds shipping_info to available product feed attributes.
+   *
+   * @implements woo_feed_product_attribute_dropdown
+   */
+  public static function woo_feed_product_attribute_dropdown($attributes): array {
+    $group_id = 60;
+    while (isset($attributes['--' . $group_id])) {
+      $group_id++;
+    }
+    $attributes["--$group_id"] = 'Dynamic Shipping Info';
+    $attributes['dynamic_shipping_info'] = 'Dynamic Shipping Info';
+    $attributes["---$group_id"] = '';
+    return $attributes;
+  }
+
+  /**
+   * Generates a product feed value for attribute 'dynamic_shipping_info'.
+   *
+   * @implements woo_feed_get_{$attribute}_attribute
+   */
+  public static function woo_feed_get_dynamic_shipping_info_attribute($output, $product, $config): string {
```

Comment:
I am afraid that this might trigger a fatal error if `$product` is `null` as `get_product_dynamic_shipping_text` method expect a `WC_Product`. We have seen this in the past with for example orphan variations. Maybe we could check if `$product` is an instance of `WC_Product` otherwise return?

Reply:
I added type hints to enforce a WC_Product now.  I get that is the opposite of what you're saying.  It sounds odd to me that the product feed generation would ever pass null as product, as it literally loops over products to serialize them. In my opinion, if the issue occurs again, we should analyze and fix the root cause within the webappick plugin.